src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,93 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = new Object[DEFAULT_SIZE];
+        values = new Object[DEFAULT_SIZE];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+
+        for (int i = 0; i < size; i++) {
+            if (key.equals(keys[i])) {
+                values[i] = value;
+                return;
+            }
+        }
```

Comment:
There is no need to split logic for null and notNull flows. There is standard comparison pattern that is suiable for both null and notNull cases:
```suggestion
        for (int i = 0; i < size; i++) {
            if (keys == keys[i] || key != null && key.equals(keys[i])) {
                values[i] = value;
                return;
            }
        }
```
and same for get() operation